src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "./BLSRegistryCoordinatorWithIndices.sol";
+
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
+
+/**
+ * @title BLSOperatorStateRetriever with view functions that allow to retrieve the state of an AVSs registry system.
+ * @author Layr Labs Inc.
+ */
+contract BLSOperatorStateRetriever {
+    struct Operator {
+        bytes32 operatorId;
+        uint96 stake;
+    }
+
+    struct CheckSignaturesIndices {
+        uint32[] nonSignerQuorumBitmapIndices;
+        uint32[] quorumApkIndices;
+        uint32[] totalStakeIndices;  
+        uint32[][] nonSignerStakeIndices; // nonSignerStakeIndices[quorumNumberIndex][nonSignerIndex]
+    }
+
+    IBLSRegistryCoordinatorWithIndices public registryCoordinator;
+    IStakeRegistry public stakeRegistry;
+    IBLSPubkeyRegistry public blsPubkeyRegistry;
+    IIndexRegistry public indexRegistry;
+
+    constructor(IBLSRegistryCoordinatorWithIndices _registryCoordinator) {
+        registryCoordinator = _registryCoordinator;
```

Comment:
this shouldn't be an AVS-specific contract. we should just deploy one and use the registryCoordinator as an input to functions.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/85137533052f6db3db219880c3d77a7f1098fb58